Content.Server/Zombies/ZombifyOnDeathSystem.cs

```diff
@@ -0,0 +1,147 @@
+using Content.Server.Nutrition.Components;
+using Robust.Shared.Player;
+using Content.Server.Popups;
+using Content.Server.Speech.Components;
+using Content.Server.Body.Systems;
+using Content.Server.CombatMode;
+using Content.Server.Inventory;
+using Content.Server.Mind.Components;
+using Content.Server.Chat.Managers;
+using Content.Server.Ghost.Roles.Components;
+using Content.Server.Hands.Components;
+using Content.Server.Mind.Commands;
+using Robust.Shared.Prototypes;
+using Content.Shared.Actions.ActionTypes;
+
+namespace Content.Server.Zombies
+{
+    /// <summary>
+    /// Handles zombie propagation and inherent zombie traits
+    /// </summary>
+    public sealed class ZombifyOnDeathSystem : EntitySystem
+    {
+        [Dependency] private readonly PolymorphableSystem _polymorph = default!;
+        [Dependency] private readonly SharedHandsSystem _sharedHands = default!;
+        [Dependency] private readonly PopupSystem _popupSystem = default!;
+        [Dependency] private readonly BloodstreamSystem _bloodstream = default!;
+        [Dependency] private readonly ServerInventorySystem _serverInventory = default!;
+        [Dependency] private readonly DamageableSystem _damageable = default!;
+        [Dependency] private readonly SharedHumanoidAppearanceSystem _sharedHuApp = default!;
+        [Dependency] private readonly IChatManager _chatMan = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<ZombifyOnDeathComponent, DamageChangedEvent>(OnDamageChanged);
+        }
+
+        /// <summary>
+        /// Handles an entity turning into a zombie when they die or go into crit
+        /// </summary>
+        private void OnDamageChanged(EntityUid uid, ZombifyOnDeathComponent component, DamageChangedEvent args)
+        {
+            if (!TryComp<MobStateComponent>(uid, out var mobstate))
+                return;
+
+            if (mobstate.IsDead() ||
+                mobstate.IsCritical())
+            {
+                ZombifyEntity(uid);
+            }
+        }
+
+        /// <summary>
+        /// This is the general purpose function to call if you want to zombify an entity.
+        /// It handles both humanoid and nonhumanoid transformation.
+        /// </summary>
+        /// <param name="target">the entity being zombified</param>
+        public void ZombifyEntity(EntityUid target)
+        {
+            var zombiecomp = EnsureComp<ZombifyOnDeathComponent>(target);
+            if (zombiecomp.Zombified)
+                return;
+            
+            EntityUid zombie;
+            /// The reasoning here is that for regular humanoids, doing a zombie polymorph
+            /// to a random zombie type is the most interesting way to do it. For things like animals,
+            /// the polymorph system doesn't really work at all, so they just use the old system of
+            /// hacking off all of the component
```

Comment:
this is kind of sussy. what makes the polymorph system better for handling zombies

Reply:
if you're a lizard wouldnt this like not zombify you into a lizard person, i dont see how its more interesting or what was wrong with the other method (especially since the other method isnt even getting removed)